src/main/java/org/opensearch/ltr/stats/suppliers/CacheStatsOnNodeSupplier.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.ltr.stats.suppliers;
+
+import com.o19s.es.ltr.feature.store.index.Caches;
+import org.opensearch.common.cache.Cache;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Aggregate stats on the cache used by the plugin per node.
+ */
+public class CacheStatsOnNodeSupplier implements Supplier<Map<String, Map<String, Object>>> {
+    private static final String LTR_CACHE_HIT_COUNT = "hit_count";
+    private static final String LTR_CACHE_MISS_COUNT = "miss_count";
+    private static final String LTR_CACHE_EVICTION_COUNT = "eviction_count";
+    private static final String LTR_CACHE_ENTRY_COUNT = "entry_count";
+    private static final String LTR_CACHE_MEMORY_USAGE_IN_BYTES = "memory_usage_in_bytes";
+
+    private Caches caches;
+
+    public CacheStatsOnNodeSupplier(Caches caches) {
+        this.caches = caches;
+    }
+
+    @Override
+    public Map<String, Map<String, Object>> get() {
+        Map<String, Map<String, Object>> values = new HashMap<>();
+        values.put("feature", getCacheStats(caches.featureCache()));
```

Comment:
Consider using an enum for cache types instead of hardcoded strings?

Then we can refactor this method as

```
@Override
public Map<String, Map<String, Object>> get() {
    Map<String, Map<String, Object>> values = new HashMap<>();
    for (CacheType type : CacheType.values()) {
        values.put(type.getValue(), getCacheStats(getCacheForType(type)));
    }
    return Collections.unmodifiableMap(values);
}
```

Reply:
Due to the way the caching is implemented as of now, I do not see the benefit of having an enum here, as this will still require a direct mapping between the enum and the method called to get the corresponding metrics, so the code is not going to be simplified/generified:
```java
caches.featureCache()
caches.featureSetCache()
caches.modelCache()
```

However, I will pull the hard-coded strings out of the method and refactor them to static class variables.